snakefile.py

```diff
@@ -218,21 +242,35 @@ def tool(name):
 
 
 rule fastqc_raw:
-    input: os.path.join(READS_DIR, "{sample}_R{read_num}.fastq")
+    input: trim_reads_input
     output:
-        os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.zip'),
-        os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.html')
-    log: os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R{read_num}.log')
+        r1_rep = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R1_fastqc.html'),
+        r1_zip = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R1_fastqc.zip'),
+        r2_rep = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R2_fastqc.html'),
+        r2_zip = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R2_fastqc.zip') # all outputs are provided to ensure atomicity 
```

Comment:
This is a bit confusing to me.  The inputs could be one or two file names.  Why then do we have a fixed set of four outputs?